### ISSUE TITLE
Add Dependency size checker

### ISSUE THREAD

**Issue by K-Ho** [ethereum-optimism/optimism#777]
**Is your feature request related to a problem? Please describe.**
`@eth-optimism/contracts` is 85MB - and this has been an issue for Synthetix (cc @justinjmoses)

**Describe the solution you'd like**
Would like to be able to run a tool to easily check where the size of our modules are coming from and allow us to address the issue.

**Comment by smartcontracts** [comment]
[BundlePhobia](https://bundlephobia.com/result?p=@eth-optimism/contracts@0.2.9) shows that our dependencies for `@eth-optimism/contracts` are only about ~4mb. A majority of this is `ganache-core`, which we're only using because of our current state dump process. Once we move to ChugSplash we should be able to cut `ganache-core` out entirely.

Running `du -sh *` highlights the real cause of the package size:
```sh
k@kpc:~/Projects/work/optimism/optimism/packages/contracts$ du -sh *
35M	artifacts
28M	artifacts-ovm
24K	bin
136K	cache
136K	cache-ovm
8.0K	CHANGELOG.md
4.0K	codechecks.yml
800K	contracts
76K	deploy
11M	deployments
3.9M	dist
4.0K	hardhat.config.ts
4.0K	LICENSE.txt
1.8M	node_modules
4.0K	package.json
4.0K	README.md
20K	scripts
52K	src
8.0K	tasks
976K	test
4.0K	tsconfig.build.json
184K	tsconfig.build.tsbuildinfo
4.0K	tsconfig.json
4.0K	tslint.json
```

`artifacts` and `artifacts-ovm` are huge! But why? On a whim I deleted the `build-info` folders inside of each `artifacts` and `artifacts-ovm`, here's what I get now:

```sh
2.3M	artifacts
2.2M	artifacts-ovm
24K	bin
136K	cache
136K	cache-ovm
8.0K	CHANGELOG.md
4.0K	codechecks.yml
800K	contracts
76K	deploy
11M	deployments
3.9M	dist
4.0K	hardhat.config.ts
4.0K	LICENSE.txt
1.8M	node_modules
4.0K	package.json
4.0K	README.md
20K	scripts
52K	src
8.0K	tasks
976K	test
4.0K	tsconfig.build.json
184K	tsconfig.build.tsbuildinfo
4.0K	tsconfig.json
4.0K	tslint.json
```

Huge difference. So let's make sure to drop the build-info folders in the published versions of our contracts. I'll also work on getting something in place that checks the build sizes before publish and warns about large folders.

**Comment by justinjmoses** [comment]
@smartcontracts when do you think you guys could do this and publish a `0.2.10` version?

**Comment by smartcontracts** [comment]
> @smartcontracts when do you think you guys could do this and publish a `0.2.10` version?

We have a fix for this merged in #780. I think we should just be able to merge this https://github.com/ethereum-optimism/optimism/pull/789#pullrequestreview-653967273.
